Feature/advanced data

johnnyporkchops opened the pull request:
New advanced data page
-This PR Addresses:

- [x] [#1309](https://github.com/18F/fec-cms/issues/1309) - Implement front end styling changes to adv. data page/feed
- [x] [#1396](https://github.com/18F/fec-cms/issues/1396) - Implement new "Download bulk data" tab in Adv data section

- [x] [#1400 ](https://github.com/18F/fec-cms/issues/1400) - Implement content for downloadable data accordions in Adv. data tabs (content had been added and committed for review)
 
- [x] There are some very slight vertical spacing inconsistencies between an item that ends with a paragraph and an item that ends with an accordion because we have a margin-bottom setting for last-child paragraphs. **(I think we are going to let this very slight inconsistency go for now -- @jenniferthibault seems to be OK with it)**
- [ ] Still need to use PEP8 on all files

Dependent issues:
- [x] [#1406](https://github.com/18F/fec-cms/issues/1406) - Implement new page for downloading presidential public fund submission files in Wagtail 
- [x]  [#1532](https://github.com/18F/fec-cms/issues/1532) - Write introduction for "Download bulk data" section 
- [x] [#1401](https://github.com/18F/fec-cms/issues/1401) - Make and publish pages for data catalog metadata in Wagtail
- [x] [#1399](https://github.com/18F/fec-cms/issues/1399) - Make and publish pages for bulk data metadata in Wagtail

New screenshot:
![adv-data-new](https://user-images.githubusercontent.com/5572856/33843697-1603605c-de6c-11e7-908e-8c719e67fd58.gif)

xtine commented on fec/fec/static/scss/components/_accordions.scss:
```diff
@@ -26,6 +26,9 @@
 //   </li>
 // </ul>
 //
+.js-accordion {
```
`.js-*` classes are only for triggering JS functions, not for styling.

johnnyporkchops replied:
Thanks @xtine, whats the best way to add vertical spacing under the  accordions here? Put a margin on accordion-nuetral, create a new class for adv-data accordions? Any global style would effect the one place we use accordions other than this page (and of course any place we use it in the future)

xtine replied:
Because the spending section has a list of data (All disbursements, Independent expenditures, etc), they should be marked up in a list. The accordion component itself should not have native margin because that should be taken care of by the parent (slab, list, div, etc).

xtine commented on fec/fec/static/scss/components/_posts.scss:
```diff
@@ -101,6 +108,18 @@
   }
 }
 
+.post__magnifyer{
```
Check spelling of "magnifier."

johnnyporkchops replied:
I think I should change this to "magnifying-glass" to match the name of the icon

xtine commented on fec/fec/static/scss/components/_posts.scss:
```diff
@@ -101,6 +108,18 @@
+    @include u-icon-bg($magnifying-glass, $base);
+    content: '';
+    display: inline-block;
+    width: 2rem;
```
Should be in a `u()` function.

xtine commented on fec/fec/static/scss/components/_icons.scss:
```diff
@@ -72,6 +72,10 @@
   margin-left: u(1rem);
   vertical-align: text-bottom;
+
+  &.i-arrow-right{
```
Check `{` spacing.

xtine commented on fec/data/templates/advanced.jinja:
```diff
@@ -99,256 +99,654 @@
+              <h4 class="post__magnifyer u-padding--bottom"><a href="/data/loans">Loans</a></h4>
+              <h4 class="is-disabled post__magnifyer u-margin--bottom">Debts (coming soon)</h4>
+              <div class="post--child-adv-data">
+                 &nbsp;
```
Can you explain why you are using a `&nbsp;` here?

johnnyporkchops replied:
That was left over from a placeholder I had used. I have removed it.

xtine commented on fec/data/templates/advanced.jinja:
```diff
@@ -99,256 +99,654 @@
-                <h4><a href="/data/receipts/individual-contributions">Individual contributions</a></h4>
-                <p>Search individual contributions to see who is donating to candidates and committees. Search for contributors by name, location, employer and occupation.</p>
+                <h4 class="post__magnifyer"><a href="/data/receipts/individual-contributions">Individual contributions</a></h4>
+                <div class="post--child-adv-data">
```
Is there a way to name this class more generically than to use a specific name like "child-adv-data?"

johnnyporkchops replied:
Would post--child--spacious make sense?

xtine commented on fec/fec/static/scss/components/_posts.scss:
```diff
@@ -73,6 +73,13 @@
   margin-left: u(3rem);
 }
 
+.post--child--spacious {
```
I don't think post classes are appropriate for this advanced data template. Post classes should be used for instances of "posts," like dynamic feed data (updates, search results, etc).

xtine replied:
I think you can remove this now?

xtine commented on fec/fec/static/scss/components/_posts.scss:
```diff
@@ -101,6 +108,18 @@
   }
 }
 
+.post__magnifying-glass{
```
This icon class should not be post specific, and should be in `_icons.scss` like how `.i-all-files` was defined for the previous version.

johnnyporkchops replied:
_posts.scss defines icons for posts, which is what I was using in order to use an existing class for adv data which is why I put it here.
pages:https://github.com/18F/fec-cms/blob/966993e8ba33ad5bd4e015a696b3b7a1341fadec/fec/fec/static/scss/components/_posts.scss#L99-L109

xtine commented on fec/fec/static/scss/components/_icons.scss:
```diff
@@ -72,6 +72,10 @@
   vertical-align: text-bottom;
+
+  &.i-arrow-right {
+    vertical-align: baseline;
```
What does this do?

xtine replied:
I mean, why are you using `vertical-align: baseline` here?

johnnyporkchops commented:
@patphongs , yes that wrapping on the list--flat bummed me out too. Thats tricky because of the way the list items float etc. but I maybe we can do it without having to use add'l container as you mentioned.

xtine commented on fec/data/templates/advanced.jinja:
```diff
@@ -99,256 +99,635 @@
-                <h3><i class="icon i-all-files icon--inline--left"></i><a href="/data/receipts/">All receipts</a></h3>
+            <h3>Search or browse data</h3>
+            <div class="content__section--ruled t-sans">
+              <h4 class="icon--hanging i-magnifying-glass"><a href="/data/receipts/">All receipts</a></h4>
```
I would highly recommend marking up each section as a list item.

xtine replied:
Also, you don't need to define `.icon--hanging` because this works the same:
```
<h4><i class="icon--hanging i-magnifying-glass"></i><a href="/data/receipts/">All receipts</a></h4>
```

xtine commented on fec/data/templates/advanced.jinja:
```diff
@@ -99,256 +99,635 @@
-                <h3><i class="icon i-all-files icon--inline--left"></i><a href="/data/receipts/">All receipts</a></h3>
+            <h3>Search or browse data</h3>
+            <div class="content__section--ruled t-sans">
+              <h4 class="icon--hanging i-magnifying-glass"><a href="/data/receipts/">All receipts</a></h4>
```
You can mark this up without needing `.icon--hanging` like this:
```
<h4><i class="icon icon--inline--left i-magnifying-glass"></i><a href="/data/receipts/">All receipts</a></h4>
```

xtine replied:
So I realizing that this indent design might be new so I'm paging @jenniferthibault if this is actually a new pattern to build out.

johnnyporkchops commented:
<i class="icon icon--inline--left i-magnifying-glass"></i> Does not work

xtine commented:
I just noticed that the `advanced.jinja` is getting long, and it might be helpful to break out the templates so it's easier to maintain. What do you think about creating an `advanced` directory in templates that houses includes for each section, like `advanced/raising.jinja`, `advanced/spending.jinja`, etc?

johnnyporkchops commented:
I think that is an idea to consider when we look at possibly refactoring this page.

johnnyporkchops commented:
xtine 19 hours ago  Member
I mean, why are you using vertical-align: baseline here?
I did not add that, it was already in the code.

johnnyporkchops commented:
I have restructured the html using the new list structure and moved each section to includes. I am now working on adding the remaining content. Updated screenshot above. Download links and meta data descriptions. 
There remains a  slight vertical spacing inconsistency between an item that ends with a paragraph and an item that ends with an accordion because we have a margin-bottom setting for last-child paragraphs.

jenniferthibault commented:
@johnnyporkchops given what we talked about at the retro today (keeping a smaller PR that fits the way the issues were broken down) can you outline what's left to do before this PR is ready for review?

johnnyporkchops commented:
Does this isue have to be completed before the adv data page can go live?
https://github.com/18F/fec-cms/issues/1399

johnnyporkchops commented:
This PR is ready for review for structure and styling and I will make another PR for completing the content.

jenniferthibault commented:
Great @johnnyporkchops ! To get this fully ready for review, there are a few things you should do to  help your reviewers:
- Edit the name of the PR to remove `[WIP]`
- Edit the PR summary description to accurately reflect what issues the final state of this PR addresses, and the summary of work in contains
- Add a few recent screenshots of the main components. The gif is a great general summary, but it's hard to look at for a closer review since you can't pause it.

johnnyporkchops commented:
@jenniferthibault, I actually have already updated the PR description and it references the issues this PR addresses. I will link to the new PR once its created.

johnnyporkchops commented:
@jenniferthibault , you can save the gif and open it in the preview program on a mac and see paused screenshots of each item. It was pointed out to me that too many screenshots make the PR unnecessarily long, so I thought a gif would be a good way to encapsulate them in one place. 
![dec-13-2017 14-10-50](https://user-images.githubusercontent.com/5572856/33957529-dbbf55ea-e00f-11e7-8d2f-1231149e6ae1.gif)

xtine commented on fec/data/templates/partials/advanced/bulk-data.jinja:
```diff
@@ -0,0 +1,295 @@
+<section id="bulk-data" class="row" aria-hidden="true" role="tabpanel">
+  <h2>Download bulk data</h2>
+  <p>Sed posuere consectetur est at lobortis. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor. Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Maecenas sed diam eget risus varius blandit sit amet non magna. Sed posuere consectetur est at lobortis. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor. Duis moll ...
```
We can't merge this (unless on a special feature branch) if there is still placeholder content.

xtine commented on fec/data/templates/partials/advanced/bulk-data.jinja:
```diff
@@ -0,0 +1,295 @@
+        <i class="icon i-download icon--absolute--left"></i>
+        <div class="content__section--indent-left">
+          <ul class="list--flat-bordered">
+            <li><a href="#">2017–2018</a></li>
```
Are these links supposed to work?

xtine commented on fec/data/templates/partials/advanced/filings-reports.jinja:
```diff
@@ -0,0 +1,81 @@
+        </div>
+      </li>
+    </ul>
+    {#
```
Comment should be removed if not WIP.

xtine commented on fec/fec/static/scss/components/_icons.scss:
```diff
@@ -258,6 +270,18 @@
+    @include u-icon-bg($download, $primary);
+    content: '';
+    float: left;
+    padding: 1.5rem;
```
Needs `u()`

xtine commented on fec/fec/static/scss/layout/_layout.scss:
```diff
@@ -81,13 +81,9 @@
-  }
+//to allow a block of content to be indented to the right of an icon or other content
+.content__section--indent-left {
+  margin: 0 0 u(3rem) u(3rem);
```
This can also be written like `margin: u(0 0 3rem 3rem)`

johnnyporkchops commented:
I have removed the placeholder content for now and the comment in filings-reports.jinja. 
I was going to add the hrefs for the links in the download lists in the coming content PR. please let me know if this cannot be merged to dev without those being operable as well?

johnnyporkchops commented:
I have better updated the [above PR description](https://github.com/18F/fec-cms/pull/1603#issue-278490288) to address related/dependent PRs.

johnnyporkchops commented:
I have added download links for all accordions, header file links and corrected other links/content. 

- [ ] Missing header file links for operating expenditures (oppexp) and candidate committee linkages (ccl).
Am I correct in using these links for the header files?:
https://www.fec.gov/files/bulk-downloads/data_dictionaries/

- [ ] @paul mentioned that we might need to add a Candidate summary accordion as well with download link for these:
/bulk-downloads/2018/2018_HOUSE_SENATE_CAMPAIGNS_DOWNLOAD.csv
/bulk-downloads/2018/2018_INDEPENDENT_EXPENDITURE_DOWNLOAD.csv
/bulk-downloads/2018/2018_PAC_DOWNLOAD.csv
/bulk-downloads/2018/2018_PARTY_DOWNLOAD.csv
/bulk-downloads/2018/2018_PRESIDENTIAL_CAMPAIGNS_DOWNLOAD.csv

johnnyporkchops commented:
I have added links to all header files for everything but the bulk data tab. I am not sure if those pages have not been made yet, or if they are supposed to stay linked to the zip files they are linked to currently.

johnnyporkchops commented:
Header file and metadata page links for bulk data tab have been added. Header csv file for House/Senate current campaigns(webl) and All candidates (weball) do not seem to exist so they are currently linked to the .txt file
Also, Senate unofficial electronic filings on the filings tab is missing data files.

johnnyporkchops commented:
Now all that remains outstanding content-wise is: 
- [ ] Three Header csv files do not seem to exist. For **House/Senate current campaigns (webl)** , **All candidates (weball)** and **PAC Summary(webk)**. They are currently linked to the .txt file.
See: https://www.fec.gov/files/bulk-downloads/index.html?prefix=bulk-downloads/data_dictionaries/
- [ ] Senate unofficial electronic filings on the filings tab is still missing data files.

PaulClark2 commented:
Senate unofficial electronic filings is a dynamic search system hosted and managed by Salient. We only need to provide a link to the search system.

These pages are under construction
- Current campaigns for House and Senate file metadata
- All candidates file metadata
- PAC and party summary file metadata

johnnyporkchops commented:
Can anyone provide the correct link to unofficial electronic filings dynamic search system?

johnnyporkchops commented:
Just confirming that the links for "Header file" should be linked to the metadata pages listed as referenced in the first completion criteria in [this above comment](https://github.com/18F/fec-cms/pull/1603#issuecomment-357410808)

PaulClark2 commented:
Senate unofficial electronic filings: http://docquery.fec.gov/senate/

**These files have both metadata and header files**
Committee master

- data: https://www.fec.gov/files/bulk-downloads/2018/cm18.zip
- metadata: Committee master file metadata
- header: https://www.fec.gov/files/bulk-downloads/data_dictionaries/cm_header_file.csv

Candidate master

- data: https://www.fec.gov/files/bulk-downloads/2018/cn18.zip
- metadata: Candidate master file metadata
- header: https://www.fec.gov/files/bulk-downloads/data_dictionaries/cn_header_file.csv

Candidate committee linkage

- data:https://www.fec.gov/files/bulk-downloads/2018/ccl18.zip
- metadata: Committes master file metadata
- header: https://www.fec.gov/files/bulk-downloads/data_dictionaries/ccl_header_file.csv

Any transaction from one committee to another

- data:https://www.fec.gov/files/bulk-downloads/2018/oth.zip
- metadata: Candidate committee linkage file metadata
- header: https://www.fec.gov/files/bulk-downloads/data_dictionaries/oth_header_file.csv

Contributions to candidates from committees

- data:https://www.fec.gov/files/bulk-downloads/2018/pas2.zip
- metadata: Contributions to candidates from committees metadata
- header: https://www.fec.gov/files/bulk-downloads/data_dictionaries/pas2_header_file.csv

Contributions by individuals

- data:https://www.fec.gov/files/bulk-downloads/2018/indiv.zip
- metadata: Contributions by individuals file metadata
- header: https://www.fec.gov/files/bulk-downloads/data_dictionaries/indiv_header_file.csv

Operating expenditures

- data:https://www.fec.gov/files/bulk-downloads/2018/oppexp.zip
- metadata: Operating expenditures file metadata
- header: https://www.fec.gov/files/bulk-downloads/data_dictionaries/oppexp_header_file.csv


**These files only have metadata**
Current campaigns for House and Senate (webL)

- data: https://www.fec.gov/files/bulk-downloads/2018/webl18.zip
- metadata: Current campaigns for House and Senate file metadata

All candidates file (webALL)

- data: https://www.fec.gov/files/bulk-downloads/2018/weblall8.zip
- metadata: All candidates file metadata

PAC and party summary file (webK)

- data: https://www.fec.gov/files/bulk-downloads/2018/webk18.zip
- metadata: PAC and party summary file metadata

johnnyporkchops commented:
So I should remove the "Header file" link from the sections for House/Senate current campaigns (webl) , All candidates (weball) and PAC Summary(webk).?

PaulClark2 commented:
Yes, please do.

jenniferthibault commented on fec/data/templates/partials/advanced/candidates.jinja:
```diff
@@ -0,0 +1,83 @@
+<section id="candidates" class="row" aria-hidden="true" role="tabpanel">
+  <h2>Candidates</h2>
+  <p>Search candidates for president, Senate and House for all years by location, party and more.</p>
```
Replace this text with `Search candidate data, including money raised, money spent, cash on hand and debt.`

jenniferthibault commented on fec/data/templates/partials/advanced/candidates.jinja:
```diff
@@ -0,0 +1,83 @@
+        <i class="icon i-magnifying-glass icon--absolute--left"></i>
+        <div class="content__section--indent-left">
+          <h4><a href="/data/candidates/?has_raised_funds=true">All candidates</a></h4>
+          <p class ="u-margin--bottom">Search candidates for president, Senate and House for all years by location, party and more</p>
```
Add a period at the end of this sentence.

jenniferthibault commented on fec/data/templates/partials/advanced/historical.jinja:
```diff
@@ -0,0 +1,5 @@
+<section class="row" id="historical" aria-hidden="true" role="tabpanel">
+  <h2>Historical statistics</h2>
+  <p>Datatables that summarize campaign financial activity by filer type, electon cycle and reporting period.</p>
```
Fix typos:
- `Datatables` > `Data tables`
- `electon` > `election`

jenniferthibault commented on fec/data/templates/partials/advanced/historical.jinja:
```diff
@@ -0,0 +1,5 @@
+<section class="row" id="historical" aria-hidden="true" role="tabpanel">
+  <h2>Historical statistics</h2>
+  <p>Datatables that summarize campaign financial activity by filer type, electon cycle and reporting period.</p>
+  <a class="button button--cta button--go" href="/data/">Find statistics</a>
```
From the checklist in https://github.com/18F/fec-cms/issues/1411

this link should go to https://transition.fec.gov/press/campaign_finance_statistics.shtml

jenniferthibault commented on fec/data/templates/partials/advanced/loans-debts.jinja:
```diff
@@ -0,0 +1,19 @@
+<section class="row" id="loans-debts" aria-hidden="true" role="tabpanel">
+  <h2>Loans and debts</h2>
+  <h3>Search or browse data</h3>
+  <div class="content__section--ruled t-sans">
```
This rule needs to extend the full-width of the page to match the others, like this:

![screen shot 2018-01-31 at 10 09 17 am](https://user-images.githubusercontent.com/11636908/35630231-d5248ea0-066e-11e8-9dfd-534f67fd2253.png)

jenniferthibault commented on fec/data/templates/partials/advanced/candidates.jinja:
```diff
@@ -0,0 +1,83 @@
+        <i class="icon i-magnifying-glass icon--absolute--left"></i>
+        <div class="content__section--indent-left">
+          <h4><a href="/data/filings/?form_type=F2">Most recent Statements of Candidacy (Form 2)</a></h4>
+          <p>Statements of Candidacy (Form 2) contain basic information about individuals running for federal office, including their names, addresses and authorized campaign committees.</p>
```
Add missing p class: `<p class="u-margin--bottom">` which is making the spacing between the paragraph and accordion different from the others

jenniferthibault commented on fec/data/templates/partials/advanced/spending.jinja:
```diff
@@ -0,0 +1,97 @@
+      <li>
+        <i class="icon i-magnifying-glass icon--absolute--left"></i>
+        <div class="content__section--indent-left">
+          <h4><a href="#">Independent expenditures</a></h4>
```
Link should be `/data/independent-expenditures`

jenniferthibault commented on fec/data/templates/partials/advanced/spending.jinja:
```diff
@@ -0,0 +1,97 @@
+          <div class="js-accordion accordion--neutral" data-content-prefix="independent-expenditure">
+            <button type="button" class="js-accordion-trigger accordion__button">Download this data as .CSV to use on your own</button>
+            <div class="accordion__content">
+              <h4 class="t-upper">Independent expenditures</h4>
```
❕ Style correction: In ALL instances, 
`<h4 class="t-upper">` should be `<h6 class="t-no-rules">`

The tight top and left spacing will be corrected by changes to the `.accordion__content` styles

jenniferthibault commented on fec/fec/static/scss/components/_accordions.scss:
```diff
@@ -51,6 +51,7 @@
   @include clearfix();
   border-bottom: 1px solid $base;
   padding: u(1rem);
```
❕ Style change: Increase to `padding: u(2rem);` for better universal spacing

jenniferthibault commented on fec/data/templates/partials/advanced/spending.jinja:
```diff
@@ -0,0 +1,97 @@
+            <div class="accordion__content">
+              <h4 class="t-upper">Independent expenditures</h4>
+              <i class="icon i-download icon--absolute--left"></i>
+              <div class="content__section--indent-left">
```
❕ Styling correction: This class applied here is causing wayyyy too much space below the year list, making it really awkward to read. We still want the indent, (and we still want the space below on this class for when it's used on the larger list items!) but we might need to specify a class (or use a different approach) so that the margin-bottom is 1rem instead of 3rem here.

**Too much space below**
![screen shot 2018-01-31 at 10 44 06 am](https://user-images.githubusercontent.com/11636908/35632139-d612022a-0673-11e8-8e84-bd57aa1a1a4d.png)

jenniferthibault replied:
This is probably the trickiest change, because the class is effective on the bigger list with the search icons, but here, it's creating too much space

jenniferthibault commented on fec/data/templates/partials/advanced/bulk-data.jinja:
```diff
@@ -0,0 +1,290 @@
+      <button type="button" class="js-accordion-trigger accordion__button">Paper filed reports (.fec files)</button>
+      <div class="accordion__content">
+        <p class="icon-download--inline--left"><a href="/files/bulk-downloads/data_dictionaries/">File formats, header file and metadata</a></p>
+        <p><a href="/files/bulk-downloads/paper/">Download files by day</a></p>
```
Missing `»` arrow to the right after "day"

jenniferthibault commented on fec/data/templates/partials/advanced/bulk-data.jinja:
```diff
@@ -0,0 +1,290 @@
+          </ul>
+        </div>
+        <p class="icon-download--inline--left"><a href="files/bulk-downloads/data_dictionaries/cm_header_file.csv">Header file</a></p>
+        <p><a href="/campaign-finance-data/committee-master-file-description/">Metadata for this file></a><i class="icon icon--small i-arrow-right icon--inline--right"></i></p>
```
Remove extra `>` character after `file`

jenniferthibault commented:
@johnnyporkchops how are the changes going here? Please let me know if you have questions!

patphongs commented:
@johnnyporkchops Can you please fix this bug? When you go to this initial tab url for other `http://localhost:8000/data/advanced/?tab=other`, it shows all the tabs in one long list.

johnnyporkchops commented on fec/data/templates/partials/advanced/candidates.jinja:
```diff
@@ -0,0 +1,83 @@
+        <div class="content__section--indent-left">
+          <h4><a href="/data/candidates/president/">Presidential candidates</a></h4>
+          <p>Search presidential <span class="term" data-term="candidate" title="Click to define" tabindex="0">candidate</span> data including money raised, money spent, cash on hand and debt.</p>
+          <p><a href="press/resources-journalists/presidential-public-funding/">Find which candidates have applied for presidential public funding</a> &#187; </p>
```
`/` means at the top level so it should be fine. Works on my local and on feature.

johnnyporkchops commented:
Of @jenniferthibault's recent review change-requests. I think this is the only one oustanding:https://github.com/fecgov/fec-cms/pull/1603#discussion_r165093904

johnnyporkchops commented:
Final changes for advanced data:
- [ ] Change"metadata for this" file to "data description for this file"
- [ ] move `data description for this` file link to be sibling of under download lists UL
- [ ] add `u-margin--top` utility class to `data description for this file`
- [ ] fix links for data description files to change `metadata` to `description`
- [ ] check on need for header files w/ @PaulClark2

jenniferthibault commented on fec/data/templates/partials/advanced/spending.jinja:
```diff
@@ -0,0 +1,97 @@
+              <h6 class="t-no-rules">Electioneering communications</h4>
+              <i class="icon i-download icon--absolute--left"></i>
+              <div class="content__section--indent-left">
+                <ul class="list--flat-bordered">
```
This ul class is missing the `u-no-margin` utility that the others have. It should be `list--flat-bordered u-no-margin`. 

**Current:**
![screen shot 2018-04-10 at 9 21 33 am](https://user-images.githubusercontent.com/11636908/38559307-aeec2f50-3ca0-11e8-8a9f-029b3e058116.png)


**With utility:**
![screen shot 2018-04-10 at 9 21 17 am](https://user-images.githubusercontent.com/11636908/38559316-b35b26b8-3ca0-11e8-9e29-1fec2b569a86.png)

patphongs commented on fec/data/templates/advanced.jinja:
```diff
@@ -87,268 +97,33 @@
-              </div>
-            </div>
-          </section>
+          {% include 'partials/advanced/raising.jinja' %}
```
nice job parsing all of this into separate jinja files, really helps to clean up such a long file
